analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -25,11 +25,12 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
     public sealed record SymbolicValue
     {
         // Reuse instances to save memory. This "True" has the same semantic meaning and any other symbolic value with BoolConstraint.True constraint
-        public static readonly SymbolicValue This = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue Null = new SymbolicValue().WithConstraint(ObjectConstraint.Null);
-        public static readonly SymbolicValue NotNull = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue True = new SymbolicValue().WithConstraint(BoolConstraint.True);
-        public static readonly SymbolicValue False = new SymbolicValue().WithConstraint(BoolConstraint.False);
+        public static readonly SymbolicValue Constraintless = new();
```

Comment:
Shouldn't this class have a private constructor now? To properly enforce the correct usage.

Reply:
see https://github.com/SonarSource/sonar-dotnet/pull/6908#discussion_r1133253468 I would do it in a follow-up PR. It will be mechanical and UTs only.